src/main/webapp/query-control.js

```diff
@@ -193,3 +201,32 @@ function createStateDropdownList() {
     datalist.appendChild(optionElem);
   });
 }
+
+// On page load, show appropriate icon based on the localStorage
+function loadAppropriateIcon() {
+  const locationSettings = localStorage.getItem('locationSettings');
+  if (typeof locationSettings === 'undefined' || locationSettings === 'off') {
+    localStorage.setItem('locationSettings', 'off');
+    document.getElementById('location-on-icon').style.display = 'none';
+    document.getElementById('location-off-icon').style.display = 'inline';
+  } else {
+    localStorage.setItem('locationSettings', 'on');
+    document.getElementById('location-off-icon').style.display = 'none';
+    document.getElementById('location-on-icon').style.display = 'inline';
+  }
+}
+
+// When the icon is clicked, change the location settings to the opposite
+// and show the opposite icon
+function changeLocationSettings(icon) {
```

Comment:
The code inside this function looks very similar to the code inside loadAppropriateIcon. Maybe consider trying to set it up so that you only have to write it once and one function uses the other.

Reply:
fixed